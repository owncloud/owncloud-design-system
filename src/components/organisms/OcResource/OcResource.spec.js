import { mount } from "@vue/test-utils"

import Resource from "./OcResource.vue"

const fileResource = {
  name: "forest.jpg",
  path: "nature/forest.jpg",
  thumbnail: "https://cdn.pixabay.com/photo/2015/09/09/16/05/forest-931706_960_720.jpg",
  indicators: [],
  type: "file",
  extension: "jpg"
}
const folderResource = {
  name: "Documents",
  path: "",
  indicators: [],
  type: "folder",
}

describe("OcResource", () => {
  it("doesn't emit a click if the resource is a folder", () => {
    const wrapper = mount(Resource, {
      propsData: {
        resource: folderResource,
        targetRoute: {
          name: "tests-route",
        },
      },
    })

    wrapper.find(".oc-resource-name").trigger("click")
    expect(wrapper.emitted().click).toBeFalsy()
  })

  it("doesn't emit a click if the resource is not clickable", () => {
    const wrapper = mount(Resource, {
      propsData: {
        resource: fileResource,
        isResourceClickable: false,
      },
    })

    wrapper.find(".oc-resource-name").trigger("click")
    expect(wrapper.emitted().click).toBeFalsy()
  })

  it("emits a click", () => {
    const wrapper = mount(Resource, {
      propsData: {
        resource: fileResource,
      },
    })

    wrapper.find(".oc-resource-name").trigger("click")
    expect(wrapper.emitted().click).toBeTruthy()
  })
})
