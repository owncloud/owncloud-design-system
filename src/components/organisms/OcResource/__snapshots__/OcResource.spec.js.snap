// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcResource displays parent folder name default if calculated name is empty 1`] = `
<div class="oc-resource oc-text-overflow"><button type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-undefined oc-button-passive oc-button-passive-raw" draggable="false"><span class="oc-icon oc-icon-l oc-icon-passive oc-resource-icon oc-resource-icon-file"><!----></span></button>
  <div class="oc-resource-details oc-text-overflow"><button type="button" class="oc-text-overflow oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-undefined oc-button-passive oc-button-passive-raw" draggable="false">
      <!----> <span data-test-resource-path="example.pdf" data-test-resource-name="example.pdf" data-test-resource-type="file" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">example</span></span><span class="oc-resource-extension">.pdf</span></span>
    </button>
    <div class="oc-resource-indicators"><span class="parent-folder" style="cursor: default;"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span> <span class="text">Example parent folder name</span></span></div>
  </div>
</div>
`;
